cwms_radar_api/src/main/java/cwms/radar/data/dao/RatingMetadataDao.java

```diff
@@ -0,0 +1,244 @@
+     * @param specIdMask    The spec id mask to search for ratings.
+     *                      If null, all spec ids are searched.
+     * @param seekFieldNames   The field names to use for keyset pagination.
+     * @param seekValues    The values to use for keyset pagination.
+     *                      Size and order must match seekFieldNames except for first page.
+     *                      On first page seekValues will be null.
+     * @param pageSize  The number of records to return.
+     * @return A list of RatingMetadata objects.
+     */
+    @NotNull
+    private List<RatingMetadata> getRatingMetadata(String office, String specIdMask,
+                                                  @NotNull String[] seekFieldNames,
+                                                  @Nullable Object[] seekValues,
+                                                  int pageSize) {
+        List<RatingMetadata> retval;
+
+        AV_RATING_SPEC specView = AV_RATING_SPEC.AV_RATING_SPEC;
+        AV_RATING ratView = AV_RATING.AV_RATING;
+
+        // Does it matter for speed or other reasons if we check
+        // LOC_ALIAS_CATEGORY or LOC_ALIAS_GROUP too?
+        // I know that specView ALIASED_ITEM can be null without ratView ALIASED_ITEM being null
+        // but is ALIASED_ITEM ever null when the LOC_ALIAS_* fields aren't?
+        Condition condition = specView.ALIASED_ITEM.isNull()
+                .and(ratView.ALIASED_ITEM.isNull());
+
+        if (office != null && !office.isEmpty()) {
+            condition = condition.and(specView.OFFICE_ID.eq(office));
+        }
+
+        if (specIdMask != null && !specIdMask.isEmpty()) {
+            condition = condition.and(JooqDao.caseInsensitiveLikeRegex(
+                    specView.RATING_ID, specIdMask));
+        }
+
+        Collection<OrderField<?>> seekFields = getOrderFields(seekFieldNames);
+
+        SelectSeekStepN<Record> records = dsl.select(specView.OFFICE_ID, specView.TEMPLATE_ID,
+                        specView.RATING_ID, specView.LOCATION_ID, specView.DESCRIPTION,
+                        specView.VERSION, specView.DEP_ROUNDING_SPEC, specView.IND_ROUNDING_SPECS,
+                        specView.DATE_METHODS, specView.AUTO_MIGRATE_EXT_FLAG,
+                        specView.AUTO_ACTIVATE_FLAG, specView.AUTO_UPDATE_FLAG,
+                        specView.ACTIVE_FLAG, specView.SOURCE_AGENCY, specView.ALIASED_ITEM,
+                        ratView.OFFICE_ID, ratView.RATING_ID, ratView.EFFECTIVE_DATE,
+                        ratView.CREATE_DATE, ratView.TRANSITION_DATE,
+                        ratView.VERSION, ratView.DESCRIPTION,
+                        ratView.DATABASE_UNITS, ratView.NATIVE_UNITS,
+                        ratView.ACTIVE_FLAG,
+                        ratView.FORMULA, ratView.ALIASED_ITEM
+                )
+                .from(specView)
+                .leftJoin(ratView)
+                .on(specView.RATING_ID.eq(ratView.RATING_ID)
```

Comment:
put the office ID first. it's super weird but it does tend to matter what order certain columns are in. Gives the planner a hint. otherwise it *may* try to get all the ratings and then eliminate by office. but we already know the by office set of ratings is smaller than the total set of ratings.